nixos/modules/system/boot/loader/grub/grub.nix

```diff
@@ -712,9 +712,9 @@ in
 
     (mkIf cfg.enable {
 
-      boot.loader.grub.devices = optional (cfg.device != "") cfg.device;
+      boot.loader.grub.devices = mkIf (cfg.device != "") [ cfg.device ];
```

Comment:
I am inclined to also backport this to stable because it's such a foot gun.

Reply:
I don't have a good sense of whether it will break back-compat since the priorities are different. I would probably wait a week or to before back-porting this.

Reply:
Ok. I am already triggering the backport pr, so we don't forget.